contracts/NodeDelegator.sol

```diff
@@ -338,6 +395,167 @@ contract NodeDelegator is INodeDelegator, LRTConfigRoleChecker, PausableUpgradea
+    /// Transfers the withdrawn liquid staking tokens to the `LRTDepositPool` contract.
+    /// Is only callable by accounts with the Operator role.
+    /// @param withdrawal the `withdrawal` data emitted in the `WithdrawalQueued` event from EigenLayer's
+    /// `DelegationManager` contract when `requestInternalWithdrawal` was called
+    /// on the `NodeDelegator` contract by a Prime Operator.
+    /// @return asset address of the liquid staking tokens (LST) that were claimed.
+    /// @return assets the amount of LSTs transferred to the `LRTDepositPool` contract.
+    function claimInternalWithdrawal(IDelegationManager.Withdrawal calldata withdrawal)
+        external
+        onlyLRTOperator
+        returns (address asset, uint256 assets)
+    {
+        // Make sure not withdrawing a staker's requested withdrawal
+        bytes32 withdrawalRoot = _calculateWithdrawalRoot(withdrawal);
+        if (withdrawalRequests[withdrawalRoot] != address(0)) {
+            revert NotInternalWithdrawal();
+        }
+        // Safety check that only one strategy is being withdrawn from.
+        // This should be the case as the call to EL's DelegationManager.queueWithdrawals
+        // comes from this contract's requestInternalWithdrawal function with only one strategy.
+        if (withdrawal.strategies.length != 1) {
+            revert NotSingleStrategyWithdrawal();
+        }
+        asset = address(IStrategy(withdrawal.strategies[0]).underlyingToken());
+
+        // There can be assets sitting in this NodeDelegator contract so we need to account for those.
+        uint256 assetsBefore = IERC20(asset).balanceOf(address(this));
+
+        IERC20[] memory tokens = new IERC20[](1);
+        tokens[0] = IERC20(asset);
+
+        // Claim the previously requested withdrawal from EigenLayer
+        address delegationManagerAddress = lrtConfig.getContract(LRTConstants.EIGEN_DELEGATION_MANAGER);
+        // the 3rd middlewareTimesIndexes param is not used in the EigenLayer M2 contracts
+        IDelegationManager(delegationManagerAddress).completeQueuedWithdrawal(withdrawal, tokens, 0, true);
+
+        // Remove the pending internal withdrawal shares now that they have been claimed
+        pendingInternalShareWithdrawals[address(withdrawal.strategies[0])] -= withdrawal.shares[0];
+
+        // Calculate the amount of assets returned from the withdrawal
+        assets = IERC20(asset).balanceOf(address(this)) - assetsBefore;
+        if (assets > 0) {
+            // transfer withdrawn assets to the Deposit Pool
+            IERC20(asset).transfer(lrtConfig.getContract(LRTConstants.LRT_DEPOSIT_POOL), assets);
+        }
+    }
+
+    /// @dev Returns the keccak256 hash of `withdrawal`.
+    /// @param withdrawal the `withdrawal` data emitted in the `WithdrawalQueued` event
+    /// from EigenLayer's `DelegationManager` contract.
+    function _calculateWithdrawalRoot(IDelegationManager.Withdrawal memory withdrawal)
```

Comment:
Is this more properly a WithdrawalHash, rather than a WithdrawalRoot? There's not a tree involved anywhere, is there?

Reply:
yes, it's a bad name by EigenLayer. It's a hash of the struct